gray_dataset.py

```diff
@@ -53,3 +54,24 @@ def tensor_augmentation(batch):  # Input, Output: (2,16,64,64) size tensor
     batch_return.append(target_result)
 
     return batch_return
+
+def add_noise(batch, return_noise_level):
+    data = (batch[0]/255).float() # normalisation
+    noise_std_range = (0, 55)
+    noise = np.zeros((np.shape(data)))
+    noise_std = np.random.uniform(
+        low=noise_std_range[0],
+        high=noise_std_range[1],
+        size=np.shape(data)[0],
+    )
+    for i in range(np.shape(noise)[0]):
+        noise[i,:,:] = np.random.normal(size=np.shape(data[i,:,:]), loc=0.0, scale=noise_std[i]/255)
+    noise = torch.from_numpy(noise).float()
+    noise_std = torch.from_numpy(noise_std).float()
+
+    output = Variable(data + noise).view(16, 1, batch[0].shape[1], batch[0].shape[2])
```

Comment:
Why do you need `Variable`?

Reply:
Ok I see this is from the code below, but you don't need to do it this way I think.
For now let's keep it as is to make sure we don't break anything.